src/ecTyping.ml

```diff
@@ -2811,7 +2578,7 @@ and trans_gbinding env ue decl =
   in snd_map List.flatten (List.map_fold trans1 env decl)
 
 (* -------------------------------------------------------------------- *)
-and trans_form_or_pattern env ?mv ?ps ue pf tt =
+and trans_form_or_pattern env mode ?mv ?ps ue pf tt =
```

Comment:
I assume mode allows to locate the error (not an expression) more precisely ?

Reply:
The notion of `mode` is not new. It has been used as a parameter for the operator resolution mechanism (and so was hard-coded to `Form` in this function). Now that the function is here for translating formulas & expressions, it must be passed as a parameter. And yes, it is also used to inform the function whether an expression or formula is translated and to raise an error when a formula is not a valid expression.